src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs

```diff
@@ -396,10 +396,34 @@ private string TryReceiveRawMessage()
 
             if (message == null)
             {
-                var reason = string.IsNullOrWhiteSpace(this.clientExitErrorMessage)
-                    ? CommonResources.UnableToCommunicateToTestHost
-                    : this.clientExitErrorMessage;
-                EqtTrace.Error("Unable to receive message from testhost: {0}", reason);
+                EqtTrace.Warning("TestRequestSender: Communication channel with test host is broken.");
+                var reason = CommonResources.UnableToCommunicateToTestHost;
+                if (!string.IsNullOrWhiteSpace(this.clientExitErrorMessage))
+                {
+                    reason = this.clientExitErrorMessage;
+                }
+                else
+                {
+                    // Test host process has not exited yet. We will wait for exit to allow us gather
+                    // standard error
+                    var processWaitEvent = new ManualResetEventSlim();
+                    try
+                    {
+                        EqtTrace.Info("TestRequestSender: Waiting for test host to exit.");
+                        processWaitEvent.Wait(TimeSpan.FromSeconds(10), this.clientExitCancellationSource.Token);
```

Comment:
If in 10 sec testhost doesn't exit, what will happen to testhost process? Shouldn't we kill testhost explicitly?